halibot/halauth.py

```diff
@@ -2,15 +2,15 @@
 import logging
 from halibot.message import Message
 
-def hasPermission(perm, reply=False):
+def hasPermission(perm, reply=False, argnum=-1, key="msg"):
 	def real_dec(func):
 		def wrapper(self, *args, **kwargs):
-			msg = None
-			for i in list(args) + list(kwargs.values()):
-				if i.__class__ == Message:
-					msg = i
-					break
+			if argnum >= 0 and argnum < len(args):
```

Comment:
I think you should check for `None` here. A negative `argnum` makes sense, for example if you want to refer to the last position argument with `-1`. Also, `None` is more semantically significant of a value to not be used.

Reply:
...yep, that makes a lot more sense. I've been writing too much C lately. Default `argnum` should be `None`, and I will explicitly check against `None` to see if it has been set.